geoutils/geovector.py

```diff
@@ -146,6 +155,62 @@ def crop(
             new_vector.ds = new_vector.ds.cx[xmin:xmax, ymin:ymax]
             return new_vector
 
+    def reproject(
+        self: Vector,
+        dst_ref: gu.Raster | rio.io.DatasetReader | VectorType | gpd.GeoDataFrame | str | None = None,
+        dst_crs: CRS | str | int | None = None,
+    ) -> Vector:
+        """
+        Reproject vector to a specified CRS and, optionally, cropped to certain bounds (no cropping by default).
```

Comment:
I don't understand, there is no cropping involved in this function... All that's done is to get the CRS from that reference.
I think we should go for something simple for now and only have the `dst_crs` argument. It's easy to provide it with a Raster or Vector, as the CRS is an attribute.
Or did you have something else in mind?

Reply:
Ah yes, removed the cropping in the description!
I simply wanted to mirror `Raster.reproject()`: you can provide a `dst_ref` or `dst_crs`. To have as many features consistent between `Vector` and `Raster` as possible, and keep the general policy of all our other functions of "accepting any raster or vector as inputs", so as to avoid manipulating the georeference metadata.

Reply:
It's a detail (removing 2 lines or not), so I'll merge to advance PR #350 without merging issues.
We can discuss it further at a monthly meeting if the above was not convincing :wink: .

Reply:
It's not 2 lines, but almost 30 lines of code (and a few more of docstring) !
I'm in favor of simplicity here, especially for maintenance, tests etc. Especially since we don't know how long these tools will be useful...